### DISCUSSION TITLE
robustification of the authors construct.

### DISCUSSION THREAD

**Discussion by wolfspyre** [nunocoracao/blowfish#936] [Ideas]
I struggled harder than I care to admit to make sense of the 'author' / 'authors' constructs in blowfish... and I think part of the reason is they're different, but treated similarly in enough contexts that it's confusing...

I started teasing apart the pieces and came up with a frame for them that makes WAY MORE intuitive sense to me....

I am happy to submit this in a PR,  but wanted to post it here to start a discussion on it first...
A) it's a breaking change, and so would require iterating to 3.x ... 

B) I'm not set on the names of things, or the specific ways I implemented it... it's closer to a proof of concept to allow for discussion than a polished implementation.
the concept as I have it is:

rather than 'author' and 'authors' as constructs;  I retooled things to replace `authors` with  `contributors`...

I then added a couple more config knobs for the presentation of ( construct, badge, picture )
I adjusted the evaluation of the (show/showbadge/showpicture) params to be settable in frontmatter of a specific page.

furthermore, I extended both constructs to contain i18n-scoped byline_titles, as opposed to just `author`

so all that being said...

This is how that (currently) looks in my config and on the site:

####  `config/_default/params.yaml`:

```md
#config/_default/params.yaml
---
#...
showAuthor: false
showAuthorBadge: false
showAuthorPicture: false
showContributors: false
showContributorsBadges: false
showContributorsPictures: false
#...
section: # ie articles
  showAuthor: false
  showAuthorBadge: false
  showAuthorPicture: false
  showContributors: false
  showContributorsBadges: false
  showContributorsPictures: false
#...
```

#### `config/_default/author.yaml`:
```md
#config/_default/author.yaml
---
name: 'Hoof & Paw'
image: "/img/HnP_CircleBG_SolarizedGradient.png"
headline: "Crystalini Partum a Canis Stercore"
bio: "Crystalini Partum a Canis Stercore"
byline_title:
  - "default": "Site Admin"
  - "en": "The ART in Heart"
links:
  email:
    - "mailto:site@hoof-paw.com"
  discord:
    - "https://discord.gg/EXSnM6vuP8"
  gitlab:
    - "https://gitlab.wolfspyre.io/hoof-and-paw"
  instagram:
    - "https://instagram.com/hoof.and.paw"
  etsy:
    - "https://www.etsy.com/shop/EpoxyDreaming"
```

#### `data/contributors/atticus-skwirrelbane.json:`
```json
{
  "name": "Atticus SkwirrelBane",
  "image": "img/atticus.jpg",
  "bio": "Short-tails are the best. Just dont tell Evey I said that",
  "byline_title": [
    { "default": "dog" },
    { "en": "ShortTail" }
  ],
  "headline": "Skwirrelz... I really don't like them.",
  "social": [{}]
}
```

#### `data/contributors/evey-noblewise.json:`
```json
{
  "name": "Evey Noblewise",
  "image" : "img/evey.png",
  "bio": "princess peeveypants",
  "byline_title": [
    { "default":  "dog" },
    { "en": "LongTail" }
      ],
  "headline": "Steak, Everything, Spicy steak, Nothing, Chicken, Anything, and ice cream.",
  "social": [{}]
}
```

- [a page showing atticus](https://hoof-paw.com/articles/excessive-heat/)
  - Image of the linked page as an example:  ![atticus-example](https://github.com/nunocoracao/blowfish/assets/1680659/6ed6dab1-99fa-43d4-8ea9-7ae56006dc1c)

- [a page showing evey + siteauthor](https://hoof-paw.com/articles/danger-skwirrelz-close/)
  - Image of the linked page as an example:  ![evey-example](https://github.com/nunocoracao/blowfish/assets/1680659/8d447f47-77a5-403d-9f2c-791baf465e74)

thots?
❤️🐺W

**Comment by nunocoracao** [comment]
Thanks for creating this discussion @wolfspyre - as discussed over email - for me this would be overkill and I think it will add maintenance complexity going forward. Let's hear what others have to say.

**Comment by wolfspyre** [comment]
I agree that it introduces additional complexity.
I believe the change allows contributors to be described/badged/managed independently, allowing for a more nuanced and flexible primitive that’s intuitively distinct from the author / site-admin entity construct. 

The desired goal is to have more granular control over when and where to display the constructs… distinct naming of them is a happy side bonus.

I believe the usability upside outweighs the complexity downside… I would also love to hear others’ opinions…. Maybe someone else has a better idea :)
